### ISSUE TITLE
Ferdium cache more than 100 Gb

### ISSUE THREAD

**Issue by wadid** [ferdium/ferdium-app#1814]
### Your issue

v6.7.5-nightly.14
macOs Sonoma 14.5

I ran out of storage on my Mac. I investigated the issue and found a folder 'Service Worker' and inside that 'Cache Storage' which is 116Gb. I installed Ferdium around 2,5 months ago.

Is this expected? how to prevent that?

**Comment by pepper3k** [comment]
just a guess, but there are webapps that will store a lot in the cache, telegram is a good example, just open the ferdium cache folder and try to identify which service is using it up

in most cases it can be deleted safely, just make sure you confirm this and backup before you delete anything that might be important

under that main ferdium folder there is a partitions folder, in that folder each service has it's own folder, take a look and try to identify the offending service


also, under settings > advanced, if you scroll down, you will see a cache section, this will give you some information as well as some options:

![image](https://github.com/ferdium/ferdium-app/assets/15070086/ebbb342c-aecc-4f17-9ac7-6abdd3c185bd)


hope this helps

**Comment by wadid** [comment]
yeah basically whatsapp and telegram is what I used it with. deleted both
caches... one was actually millions of files. that took quite some time.
under advanced regarding cache size it didnt even show any number anymore

Am Mo., 17. Juni 2024 um 13:51 Uhr schrieb Pepper ***@***.***

**Comment by pepper3k** [comment]
so is this problem resolved?

If so mark it as closed.

**Comment by wadid** [comment]
But isnt that a bug? Having a telegram or whatsapp cache of more than 160gb
after 2.5 months. My system stopped running because it ran out of memory.

Also ferdium didnt even show any numbers regarding cache size under
advanced.

**Comment by pepper3k** [comment]
> But isnt that a bug? Having a telegram or whatsapp cache of more than 160gb after 2.5 months. My system stopped running because it ran out of memory. Also ferdium didnt even show any numbers regarding cache size under advanced.

It's not a bug at all, this is standard operating system behavior (like it or not). The system will give applications all the resources they want, even if it results in other applications crashing or even the operating system itself. You need to always keep an eye on your resource usage. There are things like disk quotas, but you will need to set them up. The telegram mobile app has an option to limit cache size, but i don't think the web version has this option. Not sure about whatsapp.

With regards to ferdium not showing the correct size in the settings, that might be something that can be fixed. It's possible that some max value needs to be increased or something. @vraravam should have some sort of idea.

The question is, is that behavior reproducible? Can you try filling the cache again to see if this behavior will be repeated?

**Comment by wadid** [comment]
Yeah but applications normally dont set quotas themselves or give warnings?

Because this means in 2 months my OS will crash again.

**Comment by pepper3k** [comment]
If you were downloading tons of movies the same thing would happen, operating systems still rely on the user to monitor resources.

On the other hand though, @vraravam could give us some insight on maybe creating a feature request to have a cache limit enforced by ferdium, maybe even a per service limit.

That would definitely solve the problem and help manage system resources better.

**Comment by wadid** [comment]
I rarely use the service that is filling up the cache. So that means the
cache is filling up rapidly without me doing anything, until the system
breaks down. If this is not a unique case then I think it might make sense
to have an option to put a limit.

**Comment by vraravam** [comment]
@pepper3k  - I am not actively contributing to Ferdium nowadays. Please do not tag me. There are other contributors who are also on the discord channel - where one can ask for help.
At the same time, we do welcome new contributors to the codebases for such requests.

**Comment by pepper3k** [comment]
> @pepper3k - I am not actively contributing to Ferdium nowadays. Please do not tag me. There are other contributors who are also on the discord channel - where one can ask for help. At the same time, we do welcome new contributors to the codebases for such requests.

ahh i see, i just remember you from the dark days when you lead the fork from ferdi, you really saved the day, much appreciated brother

who should i be tagging, @SpecialAro ?

**Comment by SpecialAro** [comment]
Hello 👋 

In fact we don't have a specific cache limit enforced. We rely on the user going up to the service and clicking Clear Cache or going to the general settings and clicking Clear Cache.

We could, indeed and per request, implement such feature - one that would monitor how big is the cache and clear it if it hits a cap or schedule a cache cleanup every X days or so.

Nevertheless, I don't think that is common that general browsers clear the cache automatically. Though, there are some which have a setting such as "clear cache everytime the app closes"

**Comment by MegaV0lt** [comment]
+1 for
_We could, indeed and per request, implement such feature - one that would monitor how big is the cache and clear it if it hits a cap or schedule a cache cleanup every X days or so._

**Comment by wadid** [comment]
+1 yeah sounds like a reasonable feature.

Telegram used up 1.5 GB of disk space per day, which caused my sytem to crash after 2,5 months. And then I had to investigate because I had no idea what was going on.

Also I interacted with telegram maybe only 10 times in that time frame. So not a power user or something.

**Comment by anonymoustrader** [comment]
Ferdium is by far the application / tool that writes the most to the disk of my Linux OS... do you know which is the configuration that wears out the SSD the least and has the least number of writes to the disk? For example, where you store the data on disk or in RAM, is it better to hibernate the services or not?

**Comment by ScottFreeCode** [comment]
As much as this is "normal" behavior that "in theory any application can use up as much as it wants of anything, that's just how OSes work by default and you'd address it either manually or at the OS level if you want to address it," and while browser caches do get bigger than any other typical user application I know, I'm also reasonably certain that a browser cache eating over a hundred gigabytes is _not_ a typical result of any common browser usage pattern let alone of just having any badly behaved website open! Wouldn't it be worth investigating why that doesn't happen to Telegram users on Chrome (assuming it doesn't, assuming Telegram doesn't have some unique way of breaking the browser's caching behavior for all Chromium-based browsers), confirming whether it's reproducible for other Ferdium users or only wadid, and figuring out what's making it so much worse for wadid to have Telegram on Ferdium than it is for pretty much anyone using the entire web on any other browser?
